engine-spring/src/test/java/org/camunda/bpm/engine/spring/test/TransactionRollbackTest.java

```diff
@@ -0,0 +1,182 @@
+
+  @Component
+  private static class EvaluateDmn implements JavaDelegate {
+
+    @Override
+    public void execute(DelegateExecution execution) throws Exception {
+      try {
+        getJavaDelegate().execute(execution);
+      } catch (Exception e) {
+        throw new BpmnError("DmnFailed", e.getMessage());
+      }
+    }
+
+    public JavaDelegate getJavaDelegate() {
+      return new JavaDelegate() {
+        @Override
+        public void execute(DelegateExecution execution) throws Exception {
+          execution.getProcessEngineServices().getDecisionService()
+              .evaluateDecisionByKey("does not exist")
+              .variables(Variables.createVariables())
+              .evaluate();
+        }
+      };
+    }
+  }
+
+  @RunWith(SpringJUnit4ClassRunner.class)
+  @ContextConfiguration(classes = TransactionRollbackTest.SpringWithRollback.TestConfig.class)
+  @ActiveProfiles("SpringWithRollback")
+  public static class SpringWithRollback {
+    @Configuration
+    @Profile("SpringWithRollback")
+    @ImportResource("classpath:org/camunda/bpm/engine/spring/test/junit4/springTypicalUsageTest-context.xml")
+    public static class TestConfig {
+
+      @Bean
+      public EvaluateDmn evaluateDmn() {
+        return new EvaluateDmn();
+      }
+    }
+
+    @Autowired
+    @Rule
+    public ProcessEngineRule processEngineRule;
+
+
+    @Autowired
+    public RuntimeService runtimeService;
+
+    @Test(expected = UnexpectedRollbackException.class)
+    @Deployment(resources = "org/camunda/bpm/engine/spring/test/TransactionRollback.bpmn")
+    public void failsWithRollback() throws Exception {
+      runtimeService.startProcessInstanceByKey("TransactionRollback");
+    }
+
+  }
+
+  @RunWith(SpringJUnit4ClassRunner.class)
+  @ContextConfiguration(classes = TransactionRollbackTest.SpringWithWorkaround.TestConfig.class)
+  @ActiveProfiles("SpringWithWorkaround")
+  public static class SpringWithWorkaround {
+    @Configuration
+    @Profile("SpringWithWorkaround")
+    @ImportResource("classpath:org/camunda/bpm/engine/spring/test/junit4/springTypicalUsageTest-context.xml")
+    public static class TestConfig {
+
+      @Bean
+      public JavaDelegate evaluateDmn() {
+        return new JavaDelegate() {
+
+          @Autowired
+          private LogicExecution logicExecution;
+
+          @Override
+          public void execute(DelegateExecution execution) throws Exception {
+            try {
+              logicExecution.execute(execution);
+            } catch (Exception e) {
+              throw new BpmnError("DmnFailed", e.getMessage());
+            }
+          }
+        };
+      }
+
+      public class LogicExecution implements JavaDelegate {
+        public void execute(DelegateExecution execution) throws Exception {
+          new EvaluateDmn().getJavaDelegate().execute(execution);
+        }
+      }
+
+      @Bean
+      public LogicExecution createExecutor(PlatformTransactionManager txManager) {
```

Comment:
if you are using xml based configuration, you don't have to use this kind of transaction wrapping, declare LogicExecution in your spring context as bean and put @Transactional annotation on execute method.